### PR TITLE
[librdkafka] fix find_dependency LZ4.

### PR DESCRIPTION
**Pull request by cyfdecyf**
**Describe the pull request**

- #### What does your PR fix?

Include `FindLZ4.cmake` will result following error message:

```
CMake Warning (dev) at /usr/local/lib/python3.6/dist-packages/cmake/data/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:438 (message):
  The package name passed to `find_package_handle_standard_args` (LZ4) does
  not match the name of the calling package (RdKafka).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
```

I'm following the fix in vcpkg to solve problems encounted when calling `find_package(RdKafka)`. But the fix in vcpkg does not work.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline]

Sorry I don't actually use vcpkg, so I don't know if I need to update other files.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  `Yes`

### CODE REVIEW

**Comment by LilyWangLL** [comment]
Thanks for your PR. Could you update `port-version` in **vcpkg.json**?

**Comment by cyfdecyf** [comment]
@LilyWangLL Thanks for guiding me improve the PR. Changes are pushed now. Please let me know if there are still prolems to be fixed.

**Review by dg0yt — CHANGES_REQUESTED** [review]
Even with my proposed changes, there is a major problem: The custom find module doesn't find the debug variant of lz4, due to the debug postfix in vcpkg. (Just check the debug part of `config...log` for `-- Found LZ4:...`. 
So a patch or a cmake project include is required.

File: ports/librdkafka/portfile.cmake [hunk 1]
```diff
@@ -52,7 +52,7 @@ if("lz4" IN_LIST FEATURES)
     vcpkg_replace_string(
         "${CURRENT_PACKAGES_DIR}/share/rdkafka/RdKafkaConfig.cmake"
         "find_dependency(LZ4)"
-        "include(\"\${CMAKE_CURRENT_LIST_DIR}/FindLZ4.cmake\")\n  find_dependency(LZ4)"
+        "list(APPEND CMAKE_MODULE_PATH \"\${CMAKE_CURRENT_LIST_DIR}\")\n  find_dependency(LZ4)"
```
[the remark this answers is not in the document]
Comment (cyfdecyf) [reply ports/librdkafka/portfile.cmake]: Thanks for the suggestion. I've test the change locally, it worked.

**Comment by cyfdecyf** [comment]
> The custom find module doesn't find the debug variant of lz4, due to the debug postfix in vcpkg. (Just check the debug part of `config...log` for `-- Found LZ4:...`.
> So a patch or a cmake project include is required.

Thanks for the suggestions in review.

I'm not familiar with vcpkg. The changes in this PR is what I am able to do for now, sorry for that.

**Review by LilyWangLL — APPROVED** [review]

**Comment by dg0yt** [comment]
> The custom find module doesn't find the debug variant of lz4, due to the debug postfix in vcpkg. (Just check the debug part of `config...log` for `-- Found LZ4:...`.

@LilyWangLL Is this resolved?

File: ports/librdkafka/portfile.cmake [hunk 2]
```diff
@@ -50,9 +50,9 @@ vcpkg_cmake_config_fixup(
 
 if("lz4" IN_LIST FEATURES)
     vcpkg_replace_string(
-        "${CURRENT_PACKAGES_DIR}/share/rdkafka/RdKafkaConfig.cmake"
+        "${CURRENT_PACKAGES_DIR}/share/RdKafka/RdKafkaConfig.cmake"
         "find_dependency(LZ4)"
-        "include(\"\${CMAKE_CURRENT_LIST_DIR}/FindLZ4.cmake\")\n  find_dependency(LZ4)"
+        "list(INSERT CMAKE_MODULE_PATH 0 \"\${CMAKE_CURRENT_LIST_DIR}\")\n  find_dependency(LZ4)\n  list(REMOVE_AT CMAKE_MODULE_PATH 0)"
```
Comment (strega-nil-ms) [inline ports/librdkafka/portfile.cmake:55]:
```suggestion
        [[list(INSERT CMAKE_MODULE_PATH 0 "${CMAKE_CURRENT_LIST_DIR}")
  find_dependency(LZ4)
  list(REMOVE_AT CMAKE_MODULE_PATH 0)]]
```

**Review by strega-nil-ms — APPROVED** [review]

**Comment by strega-nil-ms** [comment]
Thanks!

**Pull request merged**